src/firefly/java/edu/caltech/ipac/table/io/VoTableReader.java

```diff
@@ -72,6 +83,288 @@
         return groups.toArray(new DataGroup[groups.size()]);
     }
 
+    private static String UCD = "ucd";
+    private static String REF = "ref";
+    private static String UTYPE = "utype";
+    private static String ID = "ID";
+
+    private static String getElementAttribute(VOElement element, String attName) {
+        return element.hasAttribute(attName) ? element.getAttribute(attName) : null;
+    }
+
+
+    // root VOElement for a votable file
+    private static VOElement getVOElementFromVOTable(String location, StoragePolicy policy) {
+        try {
+            VOElementFactory voFactory =  new VOElementFactory();
+            if (policy != null) {
+                voFactory.setStoragePolicy(policy);
+            }
+           return voFactory.makeVOElement(location);
+        }  catch (SAXException|IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    // get all <RESOURCE> under VOTable root or <RESOURCE>
+    private static VOElement[] getResourceChildren(VOElement parent) {
+        return parent.getChildrenByName( "RESOURCE" );
+    }
+
+    // get all <TABLE> children under <RESOURCE>
+    private static VOElement[] getTableChildren(VOElement parent) {
+        return parent.getChildrenByName( "TABLE" );
+    }
+
+    // get all <GROUP> under <TABLE>
+    private static VOElement[] getGroupsFromTable(TableElement tableEl) {
+        return tableEl.getChildrenByName("GROUP");
+    }
+
+    // get all <LINK> under <TABLE>
+    private static LinkElement[] getLinksFromTable(TableElement tableEl) {
+        return tableEl.getLinks();
+    }
+
+    // get all <PARAM> under <TABLE>
+    private  static VOElement[] getParamsFromTable(TableElement tableEl) {
+        return tableEl.getChildrenByName("PARAM");
+    }
+
+    // get all <INFO> under <TABLE>
+    private static VOElement[] getInfosFromTable(TableElement tableEl) {
+        return tableEl.getChildrenByName("INFO");
+
+    }
+
+    // get all <FIELD> under <TABLE>
+    private static FieldElement[] getFieldsFromTable(TableElement tableEl) {
+        return tableEl.getFields();
+    }
+
+    // get all <LINK> from <FIELD>
+    private static VOElement[] getLinksFromField(FieldElement fieldEl) {
+        // VOElement -> LinkElement
+        VOElement[] voLinkAry = fieldEl.getChildrenByName("LINK");
+        VOElement[] links = new LinkElement[voLinkAry.length];
+        System.arraycopy(voLinkAry, 0, links, 0, voLinkAry.length);
+
+        return links;
+    }
+
+    // convert <GROUP>s under <TABLE> to a list of GroupInfo and add it to the associatede table (a DataGroup object)
+    private static List<GroupInfo> makeGroupInfosFromTable(TableElement tableEl, DataGroup dg) {
+        VOElement[] groupAry = getGroupsFromTable(tableEl);
```

Comment:
This is a good example of an encapsulation entry-point.  GroupInfo is a known entity to Firefly and collecting GroupInfos from a votable is an identifiable task.
I would argue that `tableEl.getChildrenByName("GROUP");` is more informative then `getGroupsFromTable(tableEl);`.